internal/state/graph/gateway_test.go

```diff
@@ -148,7 +148,7 @@ func TestBuildGateway(t *testing.T) {
 	}
 	listenerAllowedRoutes := v1beta1.Listener{
 		Name:     "listener-with-allowed-routes",
-		Hostname: (*v1beta1.Hostname)(helpers.GetStringPointer("foo.example.com")),
+		Hostname: (*v1beta1.Hostname)(helpers.GetPointer("foo.example.com")),
```

Comment:
You shouldn't need the conversion for these anymore since GetPointer returns the correct type

Reply:
I tried removing but GetPointer returns a `*string` and we need `*hostname`

Reply:
Oh I think the way I was think is if you wrap "foo.example.com" in v1beta1.Hostname(). Either way is fine.

Reply:
another way here is `helpers.GetPointer[v1beta1.Hostname]("foo.example.com")`